src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.DebugVerifier.cs

```diff
@@ -119,7 +119,30 @@ private void VerifyExpression(BoundExpression expression, bool overrideSkippedEx
                     return null;
                 }
 
-                return base.VisitCollectionExpression(node);
+                // See NullableWalker.VisitCollectionExpression.getCollectionDetails() which
+                // does not have an element type for the ImplementsIEnumerable case.
+                var hasElementType = node.CollectionTypeKind is not (CollectionExpressionTypeKind.None or CollectionExpressionTypeKind.ImplementsIEnumerable);
```

Comment:
That would be the non-generic IEnumerable case, right? So that's like converting an `object~` element type to the target element type, there's no nullability check that needs to be done there?